sdk/core/Azure.Core/src/Shared/MessagingClientDiagnostics.cs

```diff
@@ -73,44 +79,72 @@ public MessagingClientDiagnostics(string clientNamespace, string? resourceProvid
         }
 
         /// <summary>
-        ///   Attempts to extract a diagnostic id from a message's properties.
+        ///   Attempts to extract the trace context from a message's properties.
         /// </summary>
         ///
-        /// <param name="properties">The properties holding the diagnostic id.</param>
-        /// <param name="id">The value of the diagnostics identifier assigned to the event. </param>
-        ///
+        /// <param name="properties">The properties holding the trace context.</param>
+        /// <param name="traceparent">The trace parent of the message.</param>
+        /// <param name="tracestate">The trace state of the message.</param>
         /// <returns><c>true</c> if the message properties contained the diagnostic id; otherwise, <c>false</c>.</returns>
-        public static bool TryExtractDiagnosticId(IReadOnlyDictionary<string, object> properties, out string? id)
+        public static bool TryExtractTraceContext(IReadOnlyDictionary<string, object> properties, out string? traceparent, out string? tracestate)
         {
-            id = null;
+            traceparent = null;
+            tracestate = null;
 
-            if (properties.TryGetValue(DiagnosticIdAttribute, out var objectId) && objectId is string stringId)
+            if (properties.TryGetValue(DiagnosticIdAttribute, out var diagnosticId) && diagnosticId is string diagnosticIdString)
```

Comment:
suggestion:

```csharp
if (properties.TryGetValue(TraceParent, out var traceParent) && traceParent is string traceParentString)
{
    traceparent = traceParentString;
    if (properties.TryGetValue(TraceState, out object state) && state is string stateString)
    {
        tracestate = stateString;
    }
    return true;
}

if (properties.TryGetValue(DiagnosticIdAttribute, out var diagnosticId) && diagnosticId is string diagnosticIdString)
{
    traceparent = diagnosticIdString;
    return true;
}
```

there should never be a tracestate without traceparent and reading it with `Diagnostic-Id` is not necessary.